tests/ratchet/test_ratchet_receiver_chains.c

```diff
@@ -0,0 +1,223 @@
+    vsc_buffer_destroy(&pub1);
+    vsc_buffer_destroy(&priv2);
+    vsc_buffer_destroy(&pub2);
+    vsc_buffer_destroy(&priv3);
+    vsc_buffer_destroy(&pub3);
+
+    vscr_ratchet_receiver_chains_destroy(&chains);
+}
+
+void
+test__add_chain__many_chains__chains_number_should_be_limited(void) {
+    vscr_ratchet_receiver_chains_t *chains = vscr_ratchet_receiver_chains_new();
+
+    vsc_buffer_t *pub[vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS + 1];
+    vscr_ratchet_receiver_chain_t *chain_arr[vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS + 1];
+
+    for (size_t i = 0; i < vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS; i++) {
+        vsc_buffer_t *priv;
+
+        generate_raw_keypair(&priv, &pub[i]);
+
+        vsc_buffer_destroy(&priv);
+
+        vscr_ratchet_receiver_chain_t *chain = vscr_ratchet_receiver_chain_new();
+        memcpy(chain->public_key, vsc_buffer_bytes(pub[i]), sizeof(chain->public_key));
+        chain_arr[i] = chain;
+
+        vscr_ratchet_receiver_chains_add_chain(chains, chain);
+
+        TEST_ASSERT_EQUAL(chain, vscr_ratchet_receiver_chains_first_chain(chains));
+
+        for (size_t j = 0; j <= i; j++) {
+            TEST_ASSERT_EQUAL(chain_arr[j], vscr_ratchet_receiver_chains_find_chain(chains, vsc_buffer_data(pub[j])));
+        }
+    }
+
+    vsc_buffer_t *priv;
+
+    generate_raw_keypair(&priv, &pub[vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS]);
+
+    vsc_buffer_destroy(&priv);
+
+    vscr_ratchet_receiver_chain_t *chain = vscr_ratchet_receiver_chain_new();
+    memcpy(chain->public_key, vsc_buffer_bytes(pub[vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS]),
+            sizeof(chain->public_key));
+    chain_arr[vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS] = chain;
+
+    vscr_ratchet_receiver_chains_add_chain(chains, chain);
+
+    TEST_ASSERT_EQUAL(chain, vscr_ratchet_receiver_chains_first_chain(chains));
+    TEST_ASSERT_EQUAL(NULL, vscr_ratchet_receiver_chains_find_chain(chains, vsc_buffer_data(pub[0])));
+
+    for (size_t j = 1; j <= vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS; j++) {
+        TEST_ASSERT_EQUAL(chain_arr[j], vscr_ratchet_receiver_chains_find_chain(chains, vsc_buffer_data(pub[j])));
+    }
+
+    for (size_t i = 0; i < vscr_ratchet_common_hidden_MAX_RECEIVERS_CHAINS + 1; i++) {
+        vsc_buffer_destroy(&pub[i]);
+    }
+
+    vscr_ratchet_receiver_chains_destroy(&chains);
+}
+
+#endif
+
+// --------------------------------------------------------------------------
+// Entrypoint.
+// --------------------------------------------------------------------------
+int
+main(void) {
+    UNITY_BEGIN();
+
+#if TEST_DEPENDENCIES_AVAILABLE
+    RUN_TEST(test__receiver_chains__adding_chains__should_be_correct);
+    RUN_TEST(test__delete_next__adding_chains__should_be_correct);
+    RUN_TEST(test__add_chain__many_chains__chains_number_should_be_limited);
+#else
+    RUN_TEST(test__nothing__feature_disabled__must_be_ignored);
+#endif
+
+    return UNITY_END();
+}
```

Comment:
Code style: ensures that there is always at lease 1 empty line at the end of the file.